packages/waku/src/lib/plugins/vite-plugin-typegen.ts

```diff
@@ -0,0 +1,146 @@
+import type { Plugin } from 'vite';
+import { promises, readFileSync } from 'fs';
+import path from 'path';
+
+const SRC_DIR = path.resolve(process.cwd(), './src/pages');
+const OUTPUT_FILE = path.resolve(process.cwd(), './entries.gen.tsx');
+
+// Recursively collect `.tsx` files in the given directory
+const collectFiles = async (dir: string): Promise<string[]> => {
+  let results: string[] = [];
+  const files = await promises.readdir(dir, { withFileTypes: true });
+
+  for (const file of files) {
+    const fullPath = path.join(dir, file.name);
+    if (file.isDirectory()) {
+      results = results.concat(await collectFiles(fullPath));
+    } else if (file.isFile() && fullPath.endsWith('.tsx')) {
+      results.push(fullPath.replace(SRC_DIR, ''));
+    }
+  }
+  return results;
+};
+
+const srcToName = (src: string) => {
```

Comment:
Now that I have a working example, I plan to circle back to refactoring some of these variables and functions. I know that I have some duplicative string formatting going on